cloud/amazon/ec2_vpc_peer.py

```diff
@@ -0,0 +1,1544 @@
+                    },
+                    {
+                        "value": "Shaolin Allen",
+                        "key": "Name"
+                    },
+                    {
+                        "value": "development",
+                        "key": "env"
+                    }
+                ],
+                "accepter_vpc_info": {
+                    "owner_id": "12345678910",
+                    "vpc_id": "vpc-12345678",
+                    "cidr_block": "172.31.0.0/16"
+                },
+                "vpc_peering_connection_id": "pcx-12345678",
+                "requester_vpc_info": {
+                    "owner_id": "12345678910",
+                    "vpc_id": "vpc-12345678",
+                    "cidr_block": "10.100.0.0/16"
+                }
+            }
+        ]
+
+    Return:
+        Tuple (bool, bool, str, dict)
+    """
+    runit = False
+    updateit = False
+    changed = False
+    success = False
+    err_msg = ''
+    results = dict()
+    params = {
+        'VpcId': vpc_id,
+        'PeerVpcId': vpc_peer_id,
+        'DryRun': check_mode
+    }
+    if peer_owner_id:
+        params['PeerOwnerId'] = peer_owner_id
+
+    success, err_msg, results = (
+        describe_peering_connections(
+            client, params['VpcId'], params['PeerVpcId'],
+            status_codes=['active']
+        )
+    )
+    if results:
+        updateit = True
+    else:
+        runit = True
+
+    if runit:
+        success, changed, err_msg, results = (
+            runner(client, 'present', params)
+        )
+        if success and changed:
+            updateit = True
+
+    if updateit:
+        if isinstance(results, list):
+            results = convert_to_lower(results[0])
+        vpc_peering_id = results['vpc_peering_connection_id']
+        success, changed, err_msg, results = (
+            update(
+                client, vpc_peering_id, tags, accept_peer,
+                accept_with_profile, region, accepter_routes,
+                requester_routes, check_mode
+            )
+        )
+        if success:
+            err_msg = ''
+
+    results = convert_to_lower(results)
+    if results.get('tags', None):
+        results['tags'] = convert_list_of_tags(results['tags'])
+
+    return success, changed, err_msg, results
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            accepter_routes=dict(type='list'),
+            requester_routes=dict(type='list'),
+            vpc_id=dict(),
+            vpc_peer_id=dict(),
+            vpc_peering_id=dict(),
+            peer_owner_id=dict(),
+            accept_peer=dict(type='bool', default=False),
+            profile=dict(),
+            accept_with_profile=dict(),
+            resource_tags=dict(type='dict'),
+            state=dict(
+                default='present', choices=[
+                    'present', 'absent', 'accept', 'reject'
+                ]
```

Comment:
The argument spec should specify the type and are any of these required or need defaults? For example, should the list types default to `[]` rather than `None`